etc/docs/Design.md

```diff
@@ -0,0 +1,100 @@
+# Things to cover
+
+## API
+
+### High level Library API
+
+The library API should provide the following methods to the clients:
+
+* Initialize an executor instance with the following settings
+  * Limits for resources
+    
+
+* Execute a command with the supplied arguments, that returns a pointer to a `Command` object that can be later used to manipulate the underlying process
+
+### GRPC API definition
+
+The GRPC API defines the following methods:
+
+* ExecuteCommand: Starts a command with limited resources in its own namespace with the supplied arguments
+* StopExecution : Stop a certain execution 
+* ListExecutions: List all the executions the connected client has rights to
+* GetExecution: Gets metadata about a certain execution defined by an execution ID
```

Comment:
What're you going to use as "execution ID"?

Reply:
I was thinking just a standard UUID. I was thinking as well to use PID, but PID could be theoretically recycled, and also since the processes run in their own PID space, it could lead to a mismatch if used incorrectly